examples/common/tracing/TraceHandlers.cpp

```diff
@@ -73,42 +86,215 @@ class TraceOutput
     TraceStream * mStream = nullptr;
 };
 
-TraceOutput output;
+struct TraceHandlerContext
+{
+    TraceOutput * sink;
+};
 
-// TODO: Framework this into a registry of handlers for different message types.
-bool TraceDefaultHandler(const TraceEventFields & trace)
+TraceOutput gTraceOutput;
+TraceHandlerContext gTraceHandlerContext = { .sink = &gTraceOutput };
+
+std::string AsJsonKey(const std::string & key, const std::string & value, bool prepend_comma)
+{
+    return (prepend_comma ? ", "s : ""s) + "\""s + key + "\": " + value;
+}
+
+std::string AsFirstJsonKey(const std::string & key, const std::string & value)
+{
+    return AsJsonKey(key, value, /* prepend_comma = */false);
+}
+
+std::string AsNextJsonKey(const std::string & key, const std::string & value)
+{
+    return AsJsonKey(key, value, /* prepend_comma = */true);
+}
+
+std::string AsJsonString(const std::string & value)
+{
+    return "\""s + value + "\""s;
+}
+
+std::string AsJsonString(const Transport::PeerAddress * peerAddress)
+{
+    char convBuf[Transport::PeerAddress::kMaxToStringSize] = {0};
+
+    peerAddress->ToString(convBuf);
+    return convBuf;
+}
+
+std::string AsJsonBool(bool isTrue)
+{
+    return isTrue ? "true"s : "false"s;
+}
+
+std::string AsJsonHexString(const uint8_t * buf, size_t bufLen)
+{
+    // Fill a string long enough for the hex conversion, that will be overwritten
+    std::string hexBuf(2 * bufLen, '0');
+
+    Encoding::BytesToLowercaseHexBuffer(buf, bufLen, hexBuf.data(), hexBuf.size());
```

Comment:
Consider sprinkling `VerifyOrDie(CHIP_NO_ERROR == ...` around things returning CHIP_ERROR just in case, even  though we know the input we provide should always succeed.

Reply:
Will do something to make it cleaner